lib/dap/filter/geoip2.rb

```diff
@@ -0,0 +1,87 @@
+require 'maxmind/db'
+
+module Dap
+module Filter
+
+require 'dap/utils/misc'
+
+module GeoIP2Library
+  GEOIP_DIRS = [
+    File.expand_path( File.join( File.dirname(__FILE__), "..", "..", "..", "data")),
+    "/var/lib/geoip",
+    "/var/lib/geoip2"
+  ]
+  GEOIP_CITY = %W{ GeoLite2-City.mmdb }
+  GEOIP_ASN = %W{ GeoLite2-ASN.mmdb }
+
+  @@geo_city = nil
+  @@geo_asn = nil
+
+  GEOIP_DIRS.each do |d|
+    GEOIP_CITY.each do |f|
+      path = File.join(d, f)
+      if ::File.exist?(path)
+        @@geo_city = MaxMind::DB.new(path, mode: MaxMind::DB::MODE_MEMORY)
+        break
+      end
+    end
+    GEOIP_ASN.each do |f|
+      path = File.join(d, f)
+      if ::File.exist?(path)
+        @@geo_asn = MaxMind::DB.new(path, mode: MaxMind::DB::MODE_MEMORY)
+        break
+      end
+    end
+  end
+end
+
+
+#
+# Add GeoIP2 tags using the MaxMind GeoIP2::City
+#
+class FilterGeoIP2
+  include BaseDecoder
+  include GeoIP2Library
+  def decode(ip)
+    unless @@geo_city
+      raise "No MaxMind GeoIP2::City data found"
+    end
+    return unless (geo_hash = @@geo_city.get(ip))
+    return Dap::Utils::Misc.flatten_hash(geo_hash)
+  end
+end
+
+#
+# Add GeoIP2 org tags using the MaxMind GeoIP2::ASN database
+#
+class FilterGeoIP2Org
```

Comment:
rename to geo_ip2_isp to better follow the database split maxmind uses (also, we should the additional fields that come with the ISP database).

Here's the fields and naming format godap currently uses:

https://github.com/rapid7/godap/blob/a426c17c1dbb8fc1846212c206f34d1bd2d1d37a/filter/geoip2.go#L107

Reply:
This is not coming from the ISP database because I do not believe a "free" one of these is readily available.  This comes from the ASN database instead, which does.